azure-kusto-ingest/azure/kusto/ingest/base_ingest_client.py

```diff
@@ -117,12 +115,11 @@ def ingest_from_dataframe(self, df: "pandas.DataFrame", ingestion_properties: In
         if not isinstance(df, DataFrame):
             raise ValueError("Expected DataFrame instance, found {}".format(type(df)))
 
-        file_name = "df_{id}_{timestamp}_{uid}.csv.gz".format(id=id(df), timestamp=int(time.time()), uid=uuid.uuid4())
+        file_name = "df_{id}_{timestamp}_{uid}.json.gz".format(id=id(df), timestamp=int(time.time()), uid=uuid.uuid4())
         temp_file_path = os.path.join(tempfile.gettempdir(), file_name)
-
-        df.to_csv(temp_file_path, index=False, encoding="utf-8", header=False, compression="gzip")
-
-        ingestion_properties.format = DataFormat.CSV
+        with gzip.open(temp_file_path, "wb") as temp_file:
+            df.to_json(temp_file, orient="records", date_format="iso", lines=True)
+        ingestion_properties.format = DataFormat.JSON
```

Comment:
thought JSON was obsolete.
multijson?

Reply:
Is it? JSON means JSONLINES, which is why I pass lines=True.
It's more performant and smaller